### PR TITLE
feat(pay): open payments support

### PR DESCRIPTION
**Pull request by kincaidoneil**
- Resolve and pay Open Payments invoices with `invoiceUrl` option
- Add docs, schema, and validation for invoices
- Query Open Payments account in parallel with SPSP query
- Simplify some config options
- Improve test coverage to 100%
- Fix probed exchange rate mistakenly reset
- Fix sending u64 probe packet, rate would incorrectly be 1:1
- Update deps for all packages, miscellaneous fixes

### CODE REVIEW

File: packages/pay/src/controllers/max-packet.ts [hunk 1]
```diff
@@ -83,19 +83,18 @@ export class MaxPacketAmountController implements StreamController {
...
 
-  /** Have we discovered the precise max packet amount of the path? */
-  isPreciseMaxKnown(): boolean {
-    return (
-      (this.state.type === MaxPacketState.PreciseMax ||
-        this.state.type === MaxPacketState.UnknownMax) &&
-      this.verifiedPathCapacity.isPositive()
-    )
+  /** TODO */
+  getKnownMaxPacketAmount(): PositiveInt | undefined {
```
[the remark this answers is not in the document]
Comment (kincaidoneil) [reply packages/pay/src/controllers/max-packet.ts]: Good question, I renamed the methods (to `getNextMaxPacketAmount` and `getDiscoveredMaxPacketAmount`, respectively) and tried to improve the comments.

TL;DR `getNextMaxPacketAmount` is used to determine the amount of the next packet, which may be used _to discover_ the precise max packet amount, whereas `getDiscoveredMaxPacketAmount` returns the _already discovered_ limit on the max packet, if it exists.

For example, the rate probe also calls `getDiscoveredMaxPacketAmount` to decide if the rate probe is complete/if it has already discovered the max packet amount. If not, the rate probe later calls `getNextMaxPacketAmount` to decide the next packet amount to use as a probe as it's discovering the max packet amount of the path.

File: packages/pay/src/controllers/max-packet.ts [hunk 2]
```diff
@@ -83,19 +83,18 @@ export class MaxPacketAmountController implements StreamController {
           .subtract(this.verifiedPathCapacity)
           .divideCeil(Int.TWO)
           .add(this.verifiedPathCapacity) as PositiveInt
-
-      case MaxPacketState.UnknownMax:
```
[the remark this answers is not in the document]
Comment (kincaidoneil) [reply packages/pay/src/controllers/max-packet.ts:87]: If no F08 was encountered after all the initial test packets (state=`UnknownMax`) and then the rate probe tries sending a max u64 packet, that can result in weird behavior -- the exchange rate may get screwed up since it can't be greater than 1:1, for example.

File: packages/pay/src/index.ts [hunk 3]
```diff
@@ -18,49 +16,72 @@ import { RateProbe } from './controllers/rate-probe'
...
     [assetCode: string]: number
   }
+  /** Callback to set the expiration timestamp of each packet given the destination ILP address */
+  getExpiry?: (destination?: string) => Date
+  /** For testing purposes: fixed amount to deliver to the recipient, in base units */
+  amountToDeliver?: Int
+  /** For testing purposes: ILP address of the STREAM receiver to send outgoing packets. Requires `sharedSecret` */
+  destinationAddress?: string
+  /** For testing purposes: symmetric key to encrypt STREAM messages. Requires `destinationAddress` */
+  sharedSecret?: Buffer
```
[the remark this answers is not in the document]
Comment (kincaidoneil) [reply packages/pay/src/index.ts:56]: Added warning logs when any of these options are provided
